test/e2e/flowexporter_test.go

```diff
@@ -22,15 +22,58 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	v1 "k8s.io/api/core/v1"
 	networkingv1 "k8s.io/api/networking/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/vmware-tanzu/antrea/pkg/features"
 )
 
-// TestFlowExporter runs flow exporter to export flow records for flows.
-// Flows are deployed between Pods on same node.
-func TestFlowExporter(t *testing.T) {
+/* Sample output from the collector:
+IPFIX-HDR:
+  version: 10,  Message Length: 288
+  Exported Time: 1605749238 (2020-11-19 01:27:18 +0000 UTC)
+  Sequence No.: 9,  Observation Domain ID: 2134708971
+DATA SET:
+  DATA RECORD-0:
+    flowStartSeconds: 1605749227
+    flowEndSeconds: 2288912640
+    sourceIPv4Address: 10.10.0.27
+    destinationIPv4Address: 10.10.0.28
+    sourceTransportPort: 34540
+    destinationTransportPort: 5201
+    protocolIdentifier: 6
+    packetTotalCount: 1037047
+    octetTotalCount: 45371902943
+    packetDeltaCount: 410256
+    octetDeltaCount: 18018632100
+    reversePacketTotalCount: 854967
+    reverseOctetTotalCount: 44461736
+    reversePacketDeltaCount: 330362
+    reverseOctetDeltaCount: 17180264
+    sourcePodName: perftest-a
+    sourcePodNamespace: antrea-test
+    sourceNodeName: k8s-node-master
+    destinationPodName: perftest-b
+    destinationPodNamespace: antrea-test
+    destinationNodeName: k8s-node-master
+    destinationClusterIPv4: 10.103.234.179
+    destinationServicePortName: antrea-test/perftest-b:
+    ingressNetworkPolicyName: test-networkpolicy-ingress
+    ingressNetworkPolicyNamespace: antrea-test
+    egressNetworkPolicyName: test-networkpolicy-egress
+    egressNetworkPolicyNamespace: antrea-test
+
+Intra-Node: Flow record information is complete for source and destination e.g. sourcePodName, destinationPodName
+Inter-Node: Flow record from destination Node is ignored, so only flow record from the source Node has its K8s info e.g., sourcePodName, sourcePodNamespace, sourceNodeName etc.
+AntreaProxy enabled (Intra-Node): Flow record information is complete for source and destination along with K8s service info such as destinationClusterIP, destinationServicePort, destinationServicePortName etc.
+AntreaProxy enabled (Inter-Node): Flow record from destination Node is ignored, so only flow record from the source Node has its K8s info like in Inter-Node case along with K8s service info such as destinationClusterIP, destinationServicePort, destinationServicePortName etc.
+*/
+
+// TestFlowExporterIntraNodeFlows tests the case, where pods are deployed on same node and their flow information is exported as IPFIX flow records.
+func TestFlowExporterIntraNodeFlows(t *testing.T) {
```

Comment:
do you think we can use this pattern for the tests, to simplify the logic and hopefully speed up the tests:

```go
func TestFlowExporter(t *testing.T) {
    // <setup code>: create all required Pods and Services
    t.Run("IntraNodeFlows", func(t *testing.T) { ... })
    t.Run("InterNodeFlows, func(t *testing.T) { ... })
    t.Run("RemoteServiceAccess", func(t *testing.T) { ... })
    t.Run("LocalServiceAccess", func(t *testing.T) { ... })
    // ...
    // <tear-down code>
}
```
It's still logically different tests but with common setup and teardown, similar to what we do for Antrea policy tests.

see https://golang.org/pkg/testing/

Reply:
Thanks for suggestion. It is clearer and faster.